kale/marshal/utils.py

```diff
@@ -0,0 +1,84 @@
+import os
+import inspect
+
+from kale.marshal import resource_save
+from kale.marshal import resource_load
+
+KALE_DATA_DIRECTORY = ""
+KALE_DIRECTORY_FILE_NAMES = []
+
+_IPYKERNEL_LOCALS = None
+
+
+def set_kale_data_directory(path):
+    """Set the Kale data directory path."""
+    global KALE_DATA_DIRECTORY
+    KALE_DATA_DIRECTORY = path
+    # create dir if not exists
+    if not os.path.isdir(KALE_DATA_DIRECTORY):
+        os.makedirs(KALE_DATA_DIRECTORY, exist_ok=True)
+
+
+def set_kale_directory_file_names():
+    """Set the list of filenames present in the Kale data directory path."""
+    global KALE_DIRECTORY_FILE_NAMES
+    KALE_DIRECTORY_FILE_NAMES = [
+        os.path.splitext(f)[0]
+        for f in os.listdir(KALE_DATA_DIRECTORY)
+        if os.path.isfile(os.path.join(KALE_DATA_DIRECTORY, f))
+    ]
+
+
+def _get_ipykernel_locals():
```

Comment:
@elikatsis @klolos I am using this approach to make it seamless retrieving the `locals` (that is the same as `globals`) from the ipython context, that is calling either `save()` or `load()` and that, in turn, call `_get_ipykernel_locals`. I am not 100% sure this is the best solution, or that it could lead to issues. WDYT?